src/theory/bags/card_solver.cpp

```diff
@@ -0,0 +1,455 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Mudathir Mohamed
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Cardinality solver for theory of bags.
+ */
+
+#include "theory/bags/card_solver.h"
+
+#include "expr/emptybag.h"
+#include "smt/logic_exception.h"
+#include "theory/bags/inference_generator.h"
+#include "theory/bags/inference_manager.h"
+#include "theory/bags/normal_form.h"
+#include "theory/bags/solver_state.h"
+#include "theory/bags/term_registry.h"
+#include "theory/uf/equality_engine_iterator.h"
+#include "util/rational.h"
+
+using namespace std;
+using namespace cvc5::context;
+using namespace cvc5::kind;
+
+namespace cvc5 {
+namespace theory {
+namespace bags {
+
+CardSolver::CardSolver(Env& env, SolverState& s, InferenceManager& im)
+    : EnvObj(env), d_state(s), d_ig(&s, &im), d_im(im), d_bagReduction(env)
+{
+  d_nm = NodeManager::currentNM();
+  d_zero = d_nm->mkConstInt(Rational(0));
+  d_one = d_nm->mkConstInt(Rational(1));
+  d_true = d_nm->mkConst(true);
+  d_false = d_nm->mkConst(false);
+}
+
+CardSolver::~CardSolver() {}
+
+void CardSolver::reset() { d_cardGraph.clear(); }
+
+bool CardSolver::isLeaf(const Node& bag)
+{
+  Node rep = d_state.getRepresentative(bag);
+  Trace("bags-card") << "-----------------------------------------"
+                     << std::endl;
+  Trace("bags-card") << "d_cardGraph :" << std::endl;
```

Comment:
You shouldn't for a for-loop in the code to print Trace information.  Guard this block with `Trace.isOn("bags-card")`.

Reply:
Deleted the whole block which was for debugging, and should not be merged to master.
I only use auto keywords for debugging. I simply forgot to delete the code.